tests/testsuite/cargo_information/within_ws_without_lockfile/stderr.log

```diff
@@ -0,0 +1,3 @@
+    Updating `dummy-registry` index
+ Downloading crates ...
+  Downloaded my-package v0.4.1+my-package (registry `dummy-registry`)
```

Comment:
@epage What do you think about this behavior? Should we pick the `0.2` compatible version instead of the highest version?

Reply:
How are we choosing 0.4?

The presence or lack of a lockfile should be transparent to the user (with the lockfile automatically created)

Reply:
> How are we choosing 0.4?

Because we don't have the lockfile. So we picked the highest version.



> The presence or lack of a lockfile should be transparent to the user (with the lockfile automatically created)

Do you mean we should create a lockfile and pick the right version but not write it out to disk?

Reply:
If you run `cargo metadata` it will render the entire dependency tree, generating a lockfile transparently.  We should do the same.  I've not looked at how we are doing things but I'm surprised we didn't get this behavior "for free".  

Looking at it, I think we selected too low-level of a lockfile function.

Reply:
Got it. It seems I should use `ops::resolve_ws`.